### PR TITLE
Support individual bucket quotas

### PR DESCRIPTION
**Pull request by rleimenstoll**
Support setting individual bucket quotas by breaking out set_quota into a set_user_quota and set_bucket_quota.

### CODE REVIEW

File: rgwadmin/rgw.py [hunk 1]
```diff
@@ -310,18 +310,42 @@ def get_user_bucket_quota(self, uid):
...
+            quota += '&max-size-kb=%d' % max_size_kb
+        if max_objects is not None:
+            quota += '&max-objects=%d' % max_objects
+        if enabled is not None:
+            quota += '&enabled=%s' % str(enabled).lower()
+        return quota
+
+    def set_quota(self, *args, **kwargs):
+        from warnings import warn
+        warn("Deprecated in favor of set_user_quota.")
```
Comment (liammonahan) [inline rgwadmin/rgw.py]: Prefer `warn("Deprecated in favor of set_user_quota.", DeprecationWarning)`

File: rgwadmin/rgw.py [hunk 2]
```diff
@@ -310,18 +310,42 @@ def get_user_bucket_quota(self, uid):
...
+        quota = self._quota(max_size_kb=max_size_kb, max_objects=max_objects,
+                            enabled=enabled)
+        parameters = 'uid=%s&quota-type=%s%s' % (uid, quota_type, quota)
+        return self.request('put', '/%s/user?quota&format=%s&%s' %
+                            (self._admin, self._response, parameters))
+
+    def set_bucket_quota(self, uid, bucket, max_size_kb=None,
+                         max_objects=None, enabled=None):
+        parameters = 'uid=%s&bucket=%s' % (uid, bucket)
         if max_size_kb is not None:
```
Comment (dengchj) [inline rgwadmin/rgw.py]: _quota() can also be reused here.

Comment (rleimenstoll) [reply rgwadmin/rgw.py]: Definitely!

File: rgwadmin/rgw.py (hunk 1 above)
Comment (rleimenstoll) [reply rgwadmin/rgw.py]: Good catch.

**Pull request merged**